crate/operator/main.py

```diff
@@ -592,3 +593,37 @@ async def update_cratedb_resource(
                     await ensure_user_password_label(
                         core, namespace, user_spec["password"]["secretKeyRef"]["name"]
                     )
+
+
+@kopf.on.update("", "v1", "secrets", labels={LABEL_USER_PASSWORD: "true"})
+async def secret_update(
+    namespace: str,
+    name: str,
+    diff: kopf.Diff,
+    **kwargs,
+):
+    for operation, field_path, old_value, new_value in diff:
+        if field_path == ("data", "password"):
+            async with ApiClient() as api_client:
+                coapi = CustomObjectsApi(api_client)
+                core = CoreV1Api(api_client)
+
+                custom_objects = await coapi.list_namespaced_custom_object(
+                    namespace=namespace,
+                    group=API_GROUP,
+                    version="v1",
+                    plural=RESOURCE_CRATEDB,
+                )
+
+                for crate_custom_object in custom_objects["items"]:
+                    crate_custom_object_users = crate_custom_object["spec"]["users"]
+
+                    for user_spec in crate_custom_object_users:
+                        if name == user_spec["password"]["secretKeyRef"]["name"]:
+                            await update_user_password(
```

Comment:
Instead of awaiting each password update, which essentially means it's all synchronously, I think using something along the lines of what we do with the statefulsets in the create handler would make sense: https://github.com/crate/crate-operator/blob/0cc2d582976b2a21e69674c99b2c958233a07a2d/crate/operator/main.py#L309-L338

```python
kopf.register(
    fn=subhandler_partial(
        update_user_password,
        host,
        username,
        old_password,
        new_password,
    ),
    id=f"update-{crate_custom_object.metadata.name}-{user_spec.name}",
    timeout=config.BOOTSTRAP_TIMEOUT,  # Not sure about this, but the operator should give up _at some point_.
)